Fix ModuleNotFoundError: No module named 'packaging'

moreati opened the pull request:
fixes #2

moreati committed:
refactor: Remove `pip freeze` fallback, unneeded with `uv`

`uv pip list` includes information about setuptools and other "special"
packages. So there's no need for special handling.

```
➜  uv git:(issue2) ✗ uv pip list --python .venv/bin/python
➜  uv git:(issue2) ✗ uv pip list --python .venv/bin/python --format=freeze
➜  uv git:(issue2) ✗ uv pip install --python .venv/bin/python setuptools pip
Resolved 2 packages in 61ms
Installed 2 packages in 13ms
 + pip==24.3.1
 + setuptools==75.6.0
➜  uv git:(issue2) ✗ uv pip list --python .venv/bin/python --format=freeze
pip==24.3.1
setuptools==75.6.0
```

moreati committed:
refactor: Remove unused _have_pip_module()

The code that called it in _get_pip() has already been removed.

jamescooke commented:
A pretty meaningless "looks good to me" - happy to test it out on my slinger box if that would be helpful.

moreati commented:
I'm waiting for uv 0.5.6, and/or mulling ways to do `--check` mode without it.

moreati committed:
fix: state=absent fails with "error: unexpected argument '-y' found"

By default `pip uninstall` prompts for confirmation whem uninstalling a
package, `-y` is needed to bypass this. `uv pip uninstall` doesn't prompt, and
doesn't have a `-y` option.

refs #7

moreati committed:
fix: ModuleNotFoundError: No module named 'packaging'

This removes any requirement for moreati.uv.pip to import setuptools or
packaging. The same functionality is now delivered by parsing the output of
`uv pip ...`. In the case of state=absent and check mode uv 0.5.6 (released
yesterday) is required.
